enshrine-esdt-safe/src/from_sovereign/transfer_tokens.rs

```diff
@@ -107,204 +87,45 @@ pub trait TransferTokensModule:
         }
     }
 
-    fn verify_operation_tokens_issue_paid(
+    fn split_payments_for_prefix_and_fee(
         &self,
-        tokens: ManagedVec<OperationEsdtPayment<Self::Api>>,
-    ) -> bool {
+        tokens: &ManagedVec<OperationEsdtPayment<Self::Api>>,
+    ) -> (
+        ManagedVec<Self::Api, EsdtTokenPayment<Self::Api>>,
+        ManagedVec<Self::Api, OperationEsdtPayment<Self::Api>>,
+        bool,
+    ) {
         let sov_prefix = self.get_sovereign_prefix();
+        let mut sov_tokens: ManagedVec<Self::Api, EsdtTokenPayment<Self::Api>> = ManagedVec::new();
```

Comment:
`sov_tokens` should be `OperationEsdtPayment` because these will be minted.
`non_sov_tokens` should be `EsdtTokenPayment` because these will be transferred from enshrine to token-handler as payment